pkg/server/postgres/postgres.go

```diff
@@ -155,3 +165,95 @@ func (p *PostgresPlugin) DeleteDatabase(db *atlas.Database) error {
 	// delete database
 	return nil
 }
+
+func checkUsers(db *atlas.Database, sqldb *sql.DB) error {
+	//// check if users exist; verify their passwords and roles
+	dbUserMap := make(map[string]string)
+	var u string
+	//getting all users from the database
+	rows, err := sqldb.Query(`SELECT u.usename AS "User Name" FROM pg_catalog.pg_user u`)
+	if err != nil {
+		return err
+	}
+	for rows.Next() {
+		if err := rows.Scan(&u); err != nil {
+			return err
+		}
+		dbUserMap[u] = "not-visited"
+	}
+
+	users := db.Spec.Users
+
+	for _, user := range users {
+		if _, ok := dbUserMap[user.Name]; !ok {
+			err = createuser(user.Name, user.Role, user.Password, sqldb)
+			if err != nil {
+				return err
+			}
+		} else {
+			dbUserMap[user.Name] = "visited"
+			if user.Role == "admin" {
+				//roles now assigned to admin is createdb, createrole
+				_, err = sqldb.Exec(fmt.Sprintf(`alter user "%s" with %s`, user.Name, admin_role))
+				if err != nil {
+					return err
+				}
+			} else if user.Role == "read" {
+				//the existing roles with the user is not changing.
+				//Todo: the previous roles exist need to be cleared for the user
+				_, err = sqldb.Exec(fmt.Sprintf(`grant select on all tables in schema public to "%s"`, user.Name))
+				if err != nil {
+					return err
+				}
+			} else {
+				return fmt.Errorf("role for the user %s is not valid", user.Name)
+			}
+			//password is getting updating
```

Comment:
No means to compare password so just updating it

Reply:
yes